src/main/kotlin/sdksetup/SDKSetupSecondStepController.kt

```diff
@@ -0,0 +1,68 @@
+package sdksetup
+
+import com.intellij.openapi.project.Project
+import utils.appendStringByMatch
+import java.io.File
+
+class SDKSetupSecondStepController {
+
+    fun addSDKToAppBuildGradle(basePath: String, appDirectory: String, project: Project) {
+        addSDKToAppBuildGradle("$basePath/$appDirectory", project)
+    }
+
+    fun addSDKToAppBuildGradle(buildGradlePath: String, project: Project) {
+        val projectBuildGradle = File("$buildGradlePath/build.gradle")
+        var content: String = projectBuildGradle.readText()
+
+        // injecting com.onesignal:OneSignal dependency
+
+        content = content.appendStringByMatch(
+            "implementation \"androidx.appcompat:appcompat:[^']*\"",
```

Comment:
These string searches look to be checking some possible dependences that the project might have. This doesn't look like it can account for any unexpected dependency so it looks like this approach won't work with may projects.

Did some searching and it looks like there is an Intellij API to read the `build.gradle` file and modify it (`com.android.tools.idea.gradle.dsl.model.GradleBuildModel`):
https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000106044-How-to-add-dependencies-to-gradle-file-from-IntelliJ-DEA-plugin-